src/libsystemd/sd-device/sd-device.c

```diff
@@ -1429,6 +1429,69 @@ _public_ int sd_device_get_seqnum(sd_device *device, uint64_t *ret) {
         return 0;
 }
 
+static int device_shall_ignore_diskseq(sd_device *device) {
+        int r;
+
+        assert(device);
+
+        /* Accessing the property will likely initialize the device if it's not done. Otherwise we'll ignore
+         * diskseq. */
+        r = device_get_property_bool(device, "ID_IGNORE_DISKSEQ");
+        if (r >= 0)
+                return r;
+        if (r != -ENOENT)
+                return r;
+
+        /* ENOENT means one of the two following cases: a) the device is not initialized yet and diskseq
+         * support should remain disabled or b) the property is not defined and diskseq should not be ignored
+         * if the device supports it. */
+        return !device->is_initialized;
+}
+
+int device_verify_diskseq(sd_device *device, int fd) {
+        uint64_t diskseq;
+        int r;
+
+        assert(device);
+        assert(fd >= 0);
+
+        r = device_shall_ignore_diskseq(device);
+        if (r < 0)
+                return r;
+        if (r == 0) {
+                r = fd_get_diskseq(fd, &diskseq);
+                if (r < 0)
+                        return r;
```

Comment:
This part is the same as `device_get_diskseq`. Why not call that instead?

Reply:
Good point, thanks.

Reply:
Ah no, I take this back. Here we really want to make the distinction between "diskseq should be ignored" (and in this case the value stored in the db can be any value) and "diskseq is not supported by the device" (and in this case we really want to verify that this hasn't changed).

Reply:
does that make sense ?

Reply:
OK. Please ignore me in that case.